src/libraries/Common/src/Interop/Browser/Interop.JavaScript.HostObject.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+internal static partial class Interop
+{
+    internal static partial class JavaScript
+    {
+        public interface IHostObject
+        { }
+
+        public class HostObject : HostObjectBase
+        {
+            public HostObject(string hostName, params object[] _params) : base(Runtime.New(hostName, _params))
+            { }
+        }
+
+        public abstract class HostObjectBase : JSObject, IHostObject
+        {
+            protected HostObjectBase(int js_handle) : base(js_handle)
+            {
+                var result = Runtime.BindHostObject(js_handle, (int)(IntPtr)Handle, out int exception);
+                if (exception != 0)
+                    throw new JSException($"HostObject Error binding: {result.ToString()}");
```

Comment:
Do we expect such exceptions to find their way out to calling code?  We generally try to have such error strings be in .resx files, which enables them to be localized, but also potentially more easily stripped away by the linker if we decide for a particular build they're not relevant.

Reply:
The problem is that we do not know really what JavaScript will return as an error message.  I can of course put the static text in the `.resx` files but the exception text would still stand.

They could definitely find it's way from the JavaScript environment for any number of reasons and fail differently depending on the type of object being instantiated.  The Host objects are not guaranteed to exist for any number of reasons on different browser contexts.